src/payment/data/service.js

```diff
@@ -59,23 +60,10 @@ function getOrderType(productType) {
 }
 
 export function transformResults(data) {
-  const results = camelCaseObject(data) || {};
+  const results = camelCaseObject(data);
 
   const lastProduct = results.products && results.products[results.products.length - 1];
   results.orderType = getOrderType(lastProduct && lastProduct.productType);
-  results.orderTotal = Number(results.orderTotal);
```

Comment:
So all of this isn't necessary; the data coming back are numbers already.  Line 73 below - regarding null providers - is deleted in @MatthewPiatetsky 's PR, so I proactively killed it here.  Once it's gone then the whole thing basically evaporates.